features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ImageButtonMapper.kt

```diff
@@ -6,97 +6,61 @@
 import android.widget.ImageButton
-import com.datadog.android.sessionreplay.internal.recorder.GlobalBounds
 import com.datadog.android.sessionreplay.internal.recorder.MappingContext
 import com.datadog.android.sessionreplay.internal.recorder.base64.Base64Serializer
-import com.datadog.android.sessionreplay.internal.recorder.base64.ImageCompression
-import com.datadog.android.sessionreplay.internal.recorder.base64.WebPImageCompression
+import com.datadog.android.sessionreplay.internal.recorder.base64.ImageWireframeHelper
+import com.datadog.android.sessionreplay.internal.recorder.densityNormalized
 import com.datadog.android.sessionreplay.model.MobileSegment
-import com.datadog.android.sessionreplay.utils.UniqueIdentifierGenerator
 
 internal class ImageButtonMapper(
-    webPImageCompression: ImageCompression = WebPImageCompression(),
-    base64Serializer: Base64Serializer = Base64Serializer.Builder().build(),
-    uniqueIdentifierGenerator: UniqueIdentifierGenerator = UniqueIdentifierGenerator
+    private val base64Serializer: Base64Serializer,
+    private val imageWireframeHelper: ImageWireframeHelper
 ) : BaseWireframeMapper<ImageButton, MobileSegment.Wireframe>(
-    webPImageCompression = webPImageCompression,
     base64Serializer = base64Serializer,
-    uniqueIdentifierGenerator = uniqueIdentifierGenerator
+    imageWireframeHelper = imageWireframeHelper
 ) {
     override fun map(
         view: ImageButton,
         mappingContext: MappingContext
     ): List<MobileSegment.Wireframe> {
-        val resources = view.resources
-        val drawable = view.drawable?.constantState?.newDrawable(resources)
-        val id = resolveChildDrawableUniqueIdentifier(view)
-
-        if (drawable == null || id == null) return emptyList()
+        val wireframes = mutableListOf<MobileSegment.Wireframe>()
 
-        val screenDensity = mappingContext.systemInformation.screenDensity
-        val bounds = resolveViewGlobalBounds(view, screenDensity)
+        val drawable = view.drawable?.current ?: return wireframes
+        val resources = view.resources
+        val density = resources.displayMetrics.density
+        val bounds = resolveViewGlobalBounds(view, density)
 
         val (shapeStyle, border) = view.background?.resolveShapeStyleAndBorder(view.alpha)
             ?: (null to null)
 
-        val wireframes = mutableListOf<MobileSegment.Wireframe>()
-
-        // if the drawable has no width/height then there's no point trying to get a bitmap
-        if (drawable.intrinsicWidth > 0 && drawable.intrinsicHeight > 0) {
-            val imageWireframe = resolveImageWireframe(
-                view,
-                id,
-                bounds,
-                shapeStyle,
-                border,
-                drawable
-            )
-            wireframes.add(imageWireframe)
+        // add background wireframes if any
+        if (shapeStyle == null && border == null) {
+            wireframes.addAll(super.map(view, mappingContext))
```

Comment:
calling super here looks quite weird, it means you are using `super` just to handle your background as an `ImageWireframe`. Reason why we moved this logic in the `BaseWireframeMapper` is to be able to be shared with other mappers later. If you want to merge this as it is is fine but I would re - think this architecture...maybe just have a method in the `BaseWireframeMapper` -> `resolveBackgroundAsScreenshot` or even better maybe you could think to a way to resolve the background from a single method:
- in case in case it can be resolved as a `shapeStyle` and `border` you return an empty `ShapeWireframe` which lays at the bottom of your wireframes and contains these attributes
- in case you cannot resolve as a `shapeStyle` and `border` you return an `ImageWireframe` as you do now

Reply:
I've refactored this code